pkg/splunk/enterprise/finalizers_test.go

```diff
@@ -130,6 +142,61 @@ func splunkDeletionTester(t *testing.T, cr splcommon.MetaObject, delete func(spl
 	c.CheckCalls(t, "Testsplctrl.CheckForDeletion", mockCalls)
 }
 
+func splunkPVCDeletionTester(t *testing.T, cr splcommon.MetaObject, delete func(splcommon.MetaObject, splcommon.ControllerClient) (bool, error)) {
```

Comment:
Is there a requirement for new pvc tester? Can't we tweak splunkDeletionTester? Although I feel like there might be a strong reason behind the same!

Reply:
Yes, the call flow is different in both cases and cannot be merged together

Reply:
Thansk for adding this, and for now it helps us. 
Future action to us:  This DeletionTester requires an overhaul to accommodate multiple features, mainly to accommodate the caller to provide the list of function calls. Otherwise, it is hard to scale any features other than the default flows.